gh-132993: expose `HASHLIB_GIL_MINSIZE` to private extension modules

picnixz opened the pull request:
<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
gh-NNNNNN: Summary of the changes made
```

Where: gh-NNNNNN refers to the GitHub issue number.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `main`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNNNN)
```

Where: [X.Y] is the branch name, for example: [3.13].

GH-NNNNNN refers to the PR number from `main`.

-->


<!-- gh-issue-number: gh-132993 -->
* Issue: gh-132993
<!-- /gh-issue-number -->

picnixz committed:
expose `HASHLIB_GIL_MINSIZE`

picnixz committed:
use exposed `GIL_MINSIZE` constant in tests

gpshead commented on Modules/sha3module.c:
```diff
@@ -639,8 +639,10 @@ _sha3_exec(PyObject *m)
+    if (PyModule_AddStringConstant(m, "implementation", "HACL") < 0) {
+        return -1;
+    }
+    if (PyModule_AddIntConstant(m, "GIL_MINSIZE", HASHLIB_GIL_MINSIZE) < 0) {
```
in all of these extension modules I suggest exposing this as _GIL_MINSIZE with the _ prefix just in case to avoid anyone happening to think of it as an attribute to rely on.  (even though these are supposed to be internal implementation detail modules).

picnixz committed:
rename `GIL_MINSIZE` to `_GIL_MINSIZE`

picnixz committed:
only use default GIL_MINSIZE if none can be found

picnixz committed:
Merge remote-tracking branch 'upstream/main' into feat/hashlib/gil-minsize-132993

picnixz committed:
update

picnixz committed:
do not skip a test

The pure Python implementation actually calls the underlying C
functions so in a sense we may still release the GIL.
